transport-native-epoll/src/main/java/io/netty/channel/epoll/Native.java

```diff
@@ -92,6 +98,39 @@
                 // Just ignore
             }
         }
+        if (checkMusl && gnulibc() == 1) {
+            // Our binary is compiled for linking with GLIBC.
+            // Let's check that we don't have anything that looks like Musl libc in our runtime.
+            try {
+                FileInputStream fis = new FileInputStream("/proc/self/maps");
+                try {
+                    BufferedReader reader = new BufferedReader(new InputStreamReader(fis));
+                    boolean muslFound = false;
+                    String line;
+                    while ((line = reader.readLine()) != null) {
+                        if (line.contains("-musl-")) {
+                            muslFound = true;
+                            break;
+                        }
+                    }
+                    if (muslFound) {
```

Comment:
nit: we could just move the whole `throw` in the if block that is contained in the while loop and so get rid of the boolean at all.